src/SDK/MonitoringApplianceEmulator/ViewModels/TracesControlViewModel.cs

```diff
@@ -0,0 +1,272 @@
+                {
+                    this.IsSmartDetectorRunning = smartDetectorRunner.IsSmartDetectorRunning;
+                }
+            };
+        }
+
+        /// <summary>
+        /// Gets the list of supported page sizes.
+        /// </summary>
+        public List<int> SupportedPageSizes { get; }
+
+        /// <summary>
+        /// Gets or sets the pageable log to display.
+        /// </summary>
+        public IPageableLog PageableLog
+        {
+            get => this.pageableLog;
+            set
+            {
+                // Unregister from the old log
+                if (this.pageableLog != null)
+                {
+                    this.pageableLog.PropertyChanged -= this.PageableLogOnPropertyChanged;
+                }
+
+                // Set the log, and register for notifications
+                this.pageableLog = value;
+                if (this.pageableLog != null)
+                {
+                    // Make sure the log's page size matches the last user selection
+                    value.PageSize = this.PageSize;
+                    this.pageableLog.PropertyChanged += this.PageableLogOnPropertyChanged;
+                }
+
+                // Finally - fire a property changed event with empty name, this will force refresh of all bindings in the model
+                this.OnPropertyChanged(string.Empty);
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the detector is currently running.
+        /// </summary>
+        public bool IsSmartDetectorRunning
+        {
+            get => this.isSmartDetectorRunning;
+            set
+            {
+                this.isSmartDetectorRunning = value;
+                this.OnPropertyChanged();
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the name of the currently displayed log.
+        /// </summary>
+        public string CurrentLogName
+        {
+            get => this.PageableLog?.Name ?? string.Empty;
+            set => this.LoadLogTask = new ObservableTask(this.LoadLogAsync(value), this.tracer);
+        }
+
+        /// <summary>
+        /// Gets a task for tracking a log's load operation
+        /// </summary>
+        public ObservableTask LoadLogTask
+        {
+            get => this.loadLogTask;
+            private set
+            {
+                this.loadLogTask = value;
+                this.OnPropertyChanged();
+            }
+        }
+
+        /// <summary>
+        /// Gets the log names available in the log archive
+        /// </summary>
+        public ObservableCollection<string> LogNames => this.logArchive.LogNames;
+
+        /// <summary>
+        /// Gets or sets the current traces pages size.
+        /// </summary>
+        public int PageSize
+        {
+            get => this.pageSize;
+            set
+            {
+                this.pageSize = value;
```

Comment:
>this.pageSize = value; [](start = 16, length = 22)

Verify that the value is one of the supported page sizes

Reply:
no need - this is only settable from the combo box

---
In reply to: [231003949](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/13#discussion_r231003949) [](ancestors = 231003949)